### ISSUE TITLE
Add spell.broken

### ISSUE THREAD

**Issue by ShadowFoxNixill** [wizards-of-lua/wizards-of-lua#237]
A read-only boolean that returns true when the spell is broken and false while it's still active.

A spell could be broken while its lua object is still referenced by another spell (e.g. an event handler or, if #238 is implemented, a command registerer). This would be a straightforward way to check whether or not that's happened.

**Comment by Adrodoc** [comment]
For clarification: a event handler (event interceptor to be precise) can never reference a broken spell. All event interceptors are deregistered when a spell is broken.

**Comment by ShadowFoxNixill** [comment]
I mean an event handler from a different spell - e.g. spell 2 might have a reference to spell 1 somewhere while 2 is waiting for events.

The reason I mentioned an event handler is that it's the easiest way for a spell to outlast another by not disappearing after a single tick.

**Comment by Adrodoc** [comment]
There is an argument to be made for `spell.broken` not to be read-only to allow easy programmatic breaking of spells. Of course you should not be able to resurrect spells. So `spell.broken = false` should throw an error (at least if the spell is broken).

**Comment by ShadowFoxNixill** [comment]
That's true. I was thinking `spell:break()`, but I suppose there's no reason it shouldn't be `spell.broken = true`.

**Comment by Adrodoc** [comment]
The read-only aspect of this request is covered by [`Spell.alive`](https://www.wizards-of-lua.net/modules/VirtualEntity#alive).

**Comment by Adrodoc** [comment]
For spells I prefer the name `broken` (or `active`) over `alive`, because spells are usually not considered to be living.
@mkarneim Do you think we should add `Spell.broken` or `Spell.active` and deprecate `VirtualEntity.alive`?
